grafana/resource_data_source.go

```diff
@@ -356,6 +356,11 @@ source selected (via the 'type' argument).
 							Optional:    true,
 							Description: "(All) Enable TLS authentication using CA cert.",
 						},
+						"tls_configuration_method": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "(All) SSL Certificate configuration, either by ‘file-path’ or ‘file-content’.",
```

Comment:
The choices can be validated this way
```suggestion
							Description: "(All) SSL Certificate configuration, either by ‘file-path’ or ‘file-content’.",
							ValidateFunc: validation.StringInSlice([]string{"file-path", "file-content"}, false),
```

Reply:
I've fixed up the commit with this change.